### PR TITLE
 Add DDALineDrawingAlgorithm.cpp

### PR DESCRIPTION
**Pull request by piyush26c**
Add implementation of DDA (Digital Differential Analyzer) Line Drawing Algorithm in C++

### CODE REVIEW

File: code/computational_geometry/src/DDALineDrawingAlgorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    std::cout << "\nEnter Y2 : ";
+    std::cin >> y2_;
+}
+
+int main()
+{
+    DDALineDrawingAlgorithm d;
+    d.getCoordinates();
+    int gd=DETECT,gm;
+    initgraph(&gd,&gm,NULL);
```
Comment (arnavb) [inline code/computational_geometry/src/DDALineDrawingAlgorithm.cpp]: Add spaces after each comma.

Comment (arnavb) [inline code/computational_geometry/src/DDALineDrawingAlgorithm.cpp]: Also, where are these functions from?

Comment (piyush26c) [reply code/computational_geometry/src/DDALineDrawingAlgorithm.cpp]: Yeah.

Comment (piyush26c) [reply code/computational_geometry/src/DDALineDrawingAlgorithm.cpp]: Graphics Library. <graphics.h>
Should I include it implicitly?

Comment (piyush26c) [reply code/computational_geometry/src/DDALineDrawingAlgorithm.cpp]: Included Graphics Library.

**Comment by arnavb** [comment]
Please don't use `graphics.h`. As you can see in this [answer](https://stackoverflow.com/questions/7860569/how-i-can-get-and-use-the-header-file-graphics-h-in-my-c-program), the library is very old and modern alternatives exist. For those, you need to vendor them in this repository as well (although I believe some are already there).

**Comment by piyush26c** [comment]
Yeah, I will remove it.

File: code/computational_geometry/src/DDALineDrawingAlgorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    int dy = y2_ - y1_;
+    // calculate steps required for creating pixels
+    int step = abs(abs(dx) > abs(dy) ? dx : dy);
+    // calculate increment in x & y for each steps
+    float xIncrement = (dx) / (float)step;
+    float yIncrement = (dy) / (float)step;
+    // drawing  pixel for each step
+    float x = x1_;
+    float y = y1_;
+    putpixel((x), (y),GREEN);   //this putpixel is for very first pixel of the line
```
Comment (arnavb) [inline code/computational_geometry/src/DDALineDrawingAlgorithm.cpp]: Even if you don't include the library, use a more modern replacement. Also, replace `math.h` with `cmath`.

Comment (piyush26c) [reply code/computational_geometry/src/DDALineDrawingAlgorithm.cpp]: As this is the basic algorithm hence it should be implemented by using a basic function which would help the early bird learner to grasp it quickly and I will change the library for inbuilt math function.

**Comment by piyush26c** [comment]
After review, am I supposed to make any changes in the above implementation?

**Comment by piyush26c** [comment]
Do I need to make changes in above implementation?

File: code/computational_geometry/src/DDALineDrawingAlgorithm.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#include <iostream>
```
Comment (arnavb) [inline code/computational_geometry/src/DDALineDrawingAlgorithm.cpp]: The folder location of this file is incorrect; please pick a specific folder or create a new one.

**Comment by Vaibhavi24** [comment]
@piyush26c Submit the changes soon!

**Comment by piyush26c** [comment]
> @piyush26c Submit the changes soon!

Yes ma'am

**Pull request closed without merging**